### PR TITLE
Add mod-side URI Handling, invite, and state code

### PR DESCRIPTION
**Pull request by emma-miler**
Linked to https://github.com/R2Northstar/NorthstarLauncher/pull/104

### CODE REVIEW

File: Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut [hunk 1]
```diff
@@ -358,6 +352,14 @@ void function SetupComboButtonTest( var menu )
 	comboStruct.navUpButtonDisabled = true
 	comboStruct.navDownButton = file.genUpButton
 
+	headerIndex++
+	buttonIndex = 0
+	//var inviteHeader = AddComboButtonHeader( comboStruct, headerIndex, "#NS_GENERATE_INVITE_HEADER" )
+	//var joinInviteButton = AddComboButton( comboStruct, headerIndex, buttonIndex++, "#NS_JOIN_INVITE" )
+	//Hud_AddEventHandler( joinInviteButton, UIE_CLICK, AdvanceMenuEventHandler( GetMenu( "JoinInviteMenu" ) ) )
+	//var generateInviteButton = AddComboButton( comboStruct, headerIndex, buttonIndex++, "#NS_GENERATE_INVITE" )
+	//Hud_AddEventHandler( generateInviteButton, UIE_CLICK, GenerateServerInvite )
```
Comment (abarichello) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_lobby.nut:361]:
```suggestion
```
Unused code

File: Northstar.Client/mod/scripts/vscripts/ui/menu_main.nut [hunk 2]
```diff
@@ -671,3 +677,21 @@ void function OpenSinglePlayerDevMenu( var button )
...
+			isOnMainMenu = true
+			NSSetOnMainMenu(true)
+		}
+		if(NSHasStoredURI() && isOnMainMenu) {
+			ShowURIDialog()
+			NSRemoveStoredURI()
+		}
+		wait 1.0
+	}
+}
```
Comment (abarichello) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_main.nut:697]: This part is not following the project's code style 😉

File: Northstar.Client/mod/resource/northstar_client_localisation_portuguese.txt [hunk 3]
```diff
@@ -299,5 +299,22 @@ Clique em Sim se você concorda. Esta escolha pode ser alterada a qualquer momen
...
+		"NS_INVITE_GENERATE_PARTY_BUTTON"    	"Crear Invitacion de Grupo"
+		"NS_INVITE_JOIN_BUTTON"              	"Aceptar Invitacion"
+		"NS_INVITE_JOIN_HEADER"           		"Aceptar Invitacion"
+		"NS_INVITE_JOIN_BODY"             		"Te han invitado a un servidor: %1 \n Te gustaría entrar a este servidor?"
+		"NS_PARTY_INVITE_HEADER"          		"Entrar al grupo"
+		"NS_PARTY_INVITE_BODY"            		"Has sido invitado a un grupo por: %1 \n Te gustaría entrar a este grupo?"
+		"NS_INVITE_GENERATE_FAILURE_HEADER"     "No se pudo crear la invitación"
+		"NS_INVITE_GENERATE_FAILURE_BODY"	    "Ocurrió un error creando la invitación: %1"
+		"NS_INVITE_GENERATE_SUCCESS_HEADER"     "Invitacion creada"
+		"NS_INVITE_GENERATE_SUCCESS_BODY"       "La invitacion ha sido copiada al PortaPapeles"
```
Comment (abarichello) [inline Northstar.Client/mod/resource/northstar_client_localisation_portuguese.txt:318]: This is spanish, not portuguese lol

**Review by abarichello — CHANGES_REQUESTED** [review]

File: Northstar.Client/mod/resource/northstar_client_localisation_portuguese.txt (hunk 3 above)
Comment (abarichello) [inline Northstar.Client/mod/resource/northstar_client_localisation_portuguese.txt:318]:
```suggestion
		"NS_INVITE_MENU_HEADER" "Convites"
		"NS_INVITE_GENERATE_SERVER_BUTTON" "Criar convite de Servidor"
		"NS_INVITE_GENERATE_PARTY_BUTTON" "Criar convite de Grupo"
		"NS_INVITE_JOIN_BUTTON" "Aceitar convite"
		"NS_INVITE_JOIN_HEADER" "Aceitar convite"
		"NS_INVITE_JOIN_BODY" "Você recebeu um convite para entrar no servidor: %1"
		"NS_PARTY_INVITE_HEADER" "Entrar no grupo"
		"NS_PARTY_INVITE_BODY" "Você recebeu um convite do grupo: %1"
		"NS_INVITE_GENERATE_FAILURE_HEADER" "Não foi possível criar o convite"
		"NS_INVITE_GENERATE_FAILURE_BODY" "Ocorreu um erro ao criar o convite: %1"
		"NS_INVITE_GENERATE_SUCCESS_HEADER" "Convite criado"
		"NS_INVITE_GENERATE_SUCCESS_BODY" "O link de convite foi copiado"
```

**Comment by emma-miler** [comment]
Planning to rework this against current master when i have the time. Drafting this for now

**Comment by ASpoonPlaysGames** [comment]
Closing as stale, and linked PR is also closed. Feel free to reopen though if you want

**Pull request closed without merging**